src/main/kotlin/org/rust/lang/core/types/ty/TyAdt.kt

```diff
@@ -20,26 +26,39 @@ import org.rust.lang.core.types.type
 @Suppress("DataClassPrivateConstructor")
 data class TyAdt private constructor(
     val item: RsStructOrEnumItemElement,
-    val typeArguments: List<Ty>
-) : Ty(mergeFlags(typeArguments)) {
+    val typeArguments: List<Ty>,
+    val lifetimeArguments: List<Re>
+) : Ty(mergeFlags(typeArguments) or mergeFlags(lifetimeArguments)) {
 
     // This method is rarely called (in comparison with folding),
     // so we can implement it in a such inefficient way
     override val typeParameterValues: Substitution
-        get() = item.typeParameters.withIndex().associate { (i, param) ->
-            TyTypeParameter.named(param) to typeArguments.getOrElse(i) { TyUnknown }
+        get() {
+            val typeSubst = item
+                .typeParameters
+                .withIndex()
+                .associate { (i, param) ->
```

Comment:
I believe it's ok to write it in a single line:
` val typeSubst = item.typeParameters.withIndex().associate { (i, param) ->`